### PR TITLE
SC-9077 - store/use of store refactoring

### PR DESCRIPTION
**Pull request by j-jmnz**
# Short Description

## Links to Ticket and related Pull-Requests
https://ticketsystem.hpi-schul-cloud.org/browse/SC-9077

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/organisms/DatasourceRunWebuntis.vue [hunk 1]
```diff
@@ -95,7 +95,7 @@ import ModalFooterConfirm from "@components/molecules/ModalFooterConfirm";
 import FormActions from "@components/molecules/FormActions";
 import DataTable from "@components/organisms/DataTable/DataTable";
 
-import { mapGetters } from "vuex";
+import { mapState } from "vuex";
```
Comment (CeEv) [inline src/components/organisms/DatasourceRunWebuntis.vue:98]: I think in generally we should not give any componet or side a direct access to the store, like $store.state.xxx..
Any data should comsume over $store.getters.xxx..
The same for mutations.

File: src/components/organisms/DatasourceRunWebuntis.vue [hunk 2]
```diff
@@ -152,9 +152,12 @@ export default {
 		};
 	},
 	computed: {
-		...mapGetters("webuntis-metadata", {
+		...mapState("webuntis-metadata", {
```
Comment (CeEv) [inline src/components/organisms/DatasourceRunWebuntis.vue:155]: And really i think the service template in the stores is shit. :D 
But this is not a part of your PR.

**Review by CeEv — CHANGES_REQUESTED** [review]

File: src/components/organisms/DatasourceRunWebuntis.vue (hunk 1 above)
Comment (j-jmnz) [reply src/components/organisms/DatasourceRunWebuntis.vue:98]: According to vuex good practices, it is preferable to consume data by giving direct access to the store or by mapping the state to the component over having redundant getters that return the state as it is without any modification.
You can read a bit more about this over here: https://dev.to/timothyokooboh/vuex-best-practices-45dd

File: src/components/organisms/DatasourceRunWebuntis.vue (hunk 2 above)
Comment (dominikbruechner) [reply src/components/organisms/DatasourceRunWebuntis.vue:155]: We need a service template v2, which delivers us all the standard methods for custom endpoints like GET/PATCH/DELETE/POST, while sticking to good practices. 

Particulary we need per default a #loading# state.

File: src/components/organisms/DatasourceRunWebuntis.vue (hunk 1 above)
Comment (dominikbruechner) [reply src/components/organisms/DatasourceRunWebuntis.vue:98]: https://hackernoon.com/correct-and-efficient-way-to-use-vuex-part-i-wf15432eu 

Here as well.
I am talking to Cedric about this as well, after reading up on this I am also making the case for the use of MapState, if we do not need to filter data

File: src/components/organisms/DatasourceRunWebuntis.vue (hunk 2 above)
Comment (j-jmnz) [reply src/components/organisms/DatasourceRunWebuntis.vue:155]: @dominikbruechner that's the follow-up of this ticket 👍

**Review by pkleybolte — APPROVED** [review]

**Pull request merged**